test/marginCalculator.test.ts

```diff
@@ -520,66 +545,112 @@ contract('MarginCalculator', () => {
       })
 
       it('(1) Short: 1 300 put => need 15000 cUSD collateral', async () => {
-        const collateralNeeded = scaleNum(15000)
+        const collateralNeeded = createTokenAmount(15000, ctokenDecimals)
         const vault = createVault(eth300PutCUSDC.address, undefined, cusdc.address, amountOne, undefined, 0)
         const [netValue, isExcess] = await calculator.getExcessCollateral(vault)
         assert.equal(isExcess, false)
-        assert.equal(netValue.toString(), collateralNeeded)
+        assert.equal(netValue.toString(), collateralNeeded.toString())
+      })
+    })
+
+    describe('Put vault check (with high decimal collateral token) before expiry', () => {
+      const amountOne = scaleNum(1)
+
+      let put: MockOtokenInstance
+
+      before('create put with rUSD, set oracle price for rUSD', async () => {
+        // create put with rUSD as collateral + underlying
+        put = await MockOtoken.new()
+        await put.init(addressBook.address, weth.address, rusd.address, rusd.address, scaleNum(300), expiry, true)
+        const usdcPrice = scaleNum(1)
+        await oracle.setRealTimePrice(rusd.address, usdcPrice)
+      })
+
+      it('(1) Short: 1 300 put => need 300 rUSD collateral', async () => {
+        const collateralNeeded = createTokenAmount(300, rtokenDecimals)
+        const vault = createVault(put.address, undefined, rusd.address, amountOne, undefined, '0')
+        const [netValue, isExcess] = await calculator.getExcessCollateral(vault)
+        assert.equal(isExcess, false)
+        assert.equal(netValue.toString(), collateralNeeded.toString())
+      })
+
+      it('(2) Short: 1 300 put, no collateral specified => need 300 USD collateral', async () => {
+        const collateralNeeded = scaleNum(300)
+        const vault = createVault(put.address, undefined, undefined, amountOne, undefined, undefined)
+        const [netValue, isExcess] = await calculator.getExcessCollateral(vault)
+        assert.equal(isExcess, false)
+        assert.equal(netValue.toString(), collateralNeeded.toString())
+      })
+
+      it('(3) Short: 1 300 put, collateral: 350 rUSDC => excess: 50', async () => {
+        const collateralAmount = createTokenAmount(350, rtokenDecimals)
+        const expectOutPut = createTokenAmount(50, rtokenDecimals)
+        const vault = createVault(put.address, undefined, rusd.address, amountOne, undefined, collateralAmount)
+        const [netValue, isExcess] = await calculator.getExcessCollateral(vault)
+        assert.equal(isExcess, true)
+        assert.equal(netValue.toString(), expectOutPut.toString())
       })
     })
 
```

Comment:
To make sure the decimals actually work you should test something like 1.5 or 0.0004 USDC. If you test big USDC values, it won't run into the rounding to 0 issue.

Reply:
it's not an issue, it's an expected behavior to truncated decimals more than 20. But yeah just add a test for that